sklearn/utils/extmath.py

```diff
@@ -230,22 +232,22 @@ def randomized_range_finder(A, *, size, n_iter,
             Q = safe_sparse_dot(A, Q)
             Q = safe_sparse_dot(A.T, Q)
         elif power_iteration_normalizer == 'LU':
-            Q, _ = linalg.lu(safe_sparse_dot(A, Q), permute_l=True)
-            Q, _ = linalg.lu(safe_sparse_dot(A.T, Q), permute_l=True)
+            Q, _ = npx.linalg.lu(safe_sparse_dot(A, Q), permute_l=True)
+            Q, _ = npx.linalg.lu(safe_sparse_dot(A.T, Q), permute_l=True)
         elif power_iteration_normalizer == 'QR':
-            Q, _ = linalg.qr(safe_sparse_dot(A, Q), mode='economic')
-            Q, _ = linalg.qr(safe_sparse_dot(A.T, Q), mode='economic')
+            Q, _ = npx.linalg.qr(safe_sparse_dot(A, Q))
+            Q, _ = npx.linalg.qr(safe_sparse_dot(A.T, Q))
 
     # Sample the range of A using by linear projection of Q
     # Extract an orthonormal basis
-    Q, _ = linalg.qr(safe_sparse_dot(A, Q), mode='economic')
+    Q, _ = npx.linalg.qr(safe_sparse_dot(A, Q))
     return Q
 
 
 @_deprecate_positional_args
 def randomized_svd(M, n_components, *, n_oversamples=10, n_iter='auto',
                    power_iteration_normalizer='auto', transpose='auto',
-                   flip_sign=True, random_state=0):
+                   flip_sign=True, random_state=0, npx=np):
```

Comment:
actually since `randomized_svd` is a public function, it might make sense to call again `get_array_module` on `X` instead of passing `npx`.

Reply:
Yes indeed, I'll fix that